Lib/difflib.py

```diff
@@ -1634,14 +1634,18 @@ def _line_pair_iterator():
                 lines_to_write -= 1
             # Now yield the context lines after the change
             lines_to_write = context-1
-            while(lines_to_write):
-                from_line, to_line, found_diff = next(line_pair_iterator)
-                # If another change within the context, extend the context
-                if found_diff:
-                    lines_to_write = context-1
-                else:
-                    lines_to_write -= 1
-                yield from_line, to_line, found_diff
+            try:
+                while(lines_to_write):
+                    from_line, to_line, found_diff = next(line_pair_iterator)
+                    # If another change within the context, extend the context
+                    if found_diff:
+                        lines_to_write = context-1
+                    else:
+                        lines_to_write -= 1
+                    yield from_line, to_line, found_diff
+            except StopIteration:
+                # Catch exception from next() and return normally
```

Comment:
It also catches exception from "yield". Is this intentionally?

Reply:
How does *yield* raise an exception?  Also, my understanding is that *any* uncaught StopIteration in a generator is an error.

Reply:
`yield` can raise an exception passed by `throw()`.

But I agree that in this case the generator is not leaked outside, and internally it is used only for iteration.